### ISSUE TITLE
External Secret Operator stop working after update

### ISSUE THREAD

**Issue by useriu** [external-secrets/external-secrets#2540]
Hello

We are using :
Red Hat OpenShift on IBM Cloud Version: 4.11.42_1561
Kubernetes Version: v1.24.12+ceaf338
IBM Cloud Secret Manager 

We recently updated the ESO to version 0.8.3 from 0.8.2,  after that we noticed that there are no logs on external-secrets-configuration pod.
After we restarted the external-secrets-configuration pod it got on CrashLoopBackOff state. The logs for external-secrets-configuration are showing :

`panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x226167a]

goroutine 567 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x2f7dae0, 0x6276680})
	/opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:884 +0x212
github.com/external-secrets/external-secrets/pkg/provider/ibm.getImportCertSecret(0xe?, 0xc0011e1aa0, {{0xc000cc9dc0, 0x32}, {0x0, 0x0}, {0xc00135b914, 0xc}, {0x0, 0x0}, ...})
	/home/runner/work/external-secrets/external-secrets/pkg/provider/ibm/provider.go:223 +0x11a
github.com/external-secrets/external-secrets/pkg/provider/ibm.(*providerIBM).GetSecret(0xc000c7df20, {0x7fcd68195108?, 0x30?}, {{0xc000cc9dc0, 0x32}, {0x0, 0x0}, {0xc00135b914, 0xc}, {0x0, ...}, ...})
	/home/runner/work/external-secrets/external-secrets/pkg/provider/ibm/provider.go:170 +0x3c5
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).handleSecretData(_, {_, _}, _, {{{0x2ae21e4, 0xe}, {0xc0010b4aa0, 0x1b}}, {{0xc001527770, 0x17}, ...}, ...}, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller_secret.go:96 +0xd3
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).getProviderSecretData(0xc000d5b260, {0x3dd3db8, 0xc0015ceb70}, 0xc00213c800)
	/home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller_secret.go:78 +0x2be
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).Reconcile(0xc000d5b260, {0x3dd3db8, 0xc0015ceb70}, {{{0xc00135b8b0, 0xe}, {0xc001527770, 0x17}}})
	/home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_c`

This seems to block any new secret recreation from IBM Cloud Secret Manager.

We notice the same issue on a different cluster with ESO to version 0.8.3 and there we tried to update to 0.9.1 but that did not solve the issue.

Do you have any ideea what may have caused this?

**Comment by moolen** [comment]
Hey, can you share some bits of your SecretStore and the type of secret you use (looks like you're using certificate type, is there anyting else)?

Between v0.8.2 and v0.8.3 there were no changes to the IBM provider, not even dependency updates. I don't think this update caused the issue.
 
Can you share the crashlogs from 0.9.1 as well?

CC @IdanAdar

**Comment by IdanAdar** [comment]
I would compare 0.8.1 compared to 0.8.2 as that's when there were changes in the provider. 

It could be that the secret is referencing an object that doesn't exist in the provider, and so when it pulls it - it fails. If that's the case, fix the secret or the template and try again Liz

We're working on a fix that adds some more validations so that a crash won't happen.

**Comment by useriu** [comment]
Hi Team thanks for suggestion. At first we were confused because we had logs on info and there war no certificate/secret mention in the logs just what I posted.
After we switched the ESO logs to debug we were able to find the certificate which made that error.

Seems like we had define our resources wrongly, ESO expected a CA along with the  certificate  and key as we defined it but our certificate from certificate manager instance only contain a key and a certificate.

We remade the external secret to include a certificate and key without that CA certificate everything was fine.
